### PR TITLE
add missing kwargs from Enum functional api signature

### PR DESCRIPTION
**Pull request by anentropic**
I'm making an attempt to address https://github.com/python/mypy/issues/4184

```python
from enum import Enum

# Error: Too many arguments for Enum()
PictureSize = Enum('PictureSize', 'P0 P1 P2 P3 P4 P5 P6 P7 P8', type=str, module=__name__)
```

to really support the `type` arg fully I think needs https://github.com/python/typing/issues/213 ... then with the `overloads` decorator (since arg is optional) we could express that the return type is a subclass i.e. `-> Type[Intersection[_T, _T2]]`

but at the moment it's just kind of annotated and ignored

### CODE REVIEW

File: stdlib/3.4/enum.pyi [hunk 1]
```diff
@@ -18,7 +19,7 @@ class EnumMeta(ABCMeta, Iterable[Enum], Sized, Reversible[Enum], Container[Enum]
     def __members__(self: Type[_T]) -> Mapping[str, _T]: ...
 
 class Enum(metaclass=EnumMeta):
-    def __new__(cls: Type[_T], value: Any) -> _T: ...
+    def __new__(cls: Type[_T], value: str, names: Optional[str], module: Optional[str], qualname: Optional[str], type: Optional[Type[_T2]], start: Optional[int]) -> _T: ...
```
Comment (JelleZijlstra) [inline stdlib/3.4/enum.pyi]: I don't think this is right—this method is for constructing enum instances, not enum classes. If anything, your change should be on EnumMeta.

Comment (JelleZijlstra) [reply stdlib/3.4/enum.pyi]: So maybe we can't actually do much to fix this in typeshed.

Comment (anentropic) [reply stdlib/3.4/enum.pyi]: I must admit I find the `Enum` class rather baffling, I will look more into it

**Comment by anentropic** [comment]
@JelleZijlstra 

Looking into the Python source code for Enum I could see that these args actually belong on `EnumMeta.__call__`

I have added them there, however if I run mypy on my problem enum example with this modified typeshed I get:
```
.../typeshed/stdlib/3.4/enum.pyi:16: error: Signature of "__call__" incompatible with supertype "type"
```

**Comment by JelleZijlstra** [comment]
Yes, so let's close this and attempt to fix the issue in mypy instead.

**Pull request closed without merging**